clients/src/main/java/org/apache/kafka/clients/consumer/internals/Fetcher.java

```diff
@@ -611,6 +612,11 @@ private ListOffsetResult fetchOffsetsByTimes(Map<TopicPartition, Long> timestamp
                     if (records.notInitialized()) {
                         try {
                             nextInLineFetch = initializeCompletedFetch(records);
+
+                            // nextInLineRecords might be null when completedFetch contains error
+                            if (nextInLineFetch == null) {
+                                records.response.decRefCount();
```

Comment:
Can you please add a comment block to explain the different cases of how the response can be dereferenced, including
1. when an error, e.g. NOT_LEADER_FOR_PARTITION, causes the nextInLineFetch to be null, it means the underlying buffer is no longer needed
2. after we retrieve the records via the fetchRecords call below, the underlying buffer is no longer needed and it's released via the Fetcher.CompletedFetch.drain() method.